pkg/pool/pool.go

```diff
@@ -55,19 +101,12 @@ func NewWithCapacity(workers, capacity int) Interface {
 
 	// Start a go routine for each worker, which:
 	// 1. reads off of the work channel,
-	// 2. (optionally) sends errors on the error channel,
+	// 2. (optionally) sets error variable
 	// 3. marks work as done in our sync.WaitGroup.
 	for idx := 0; idx < workers; idx++ {
 		go func() {
 			for work := range i.workCh {
-				func() {
-					defer i.wg.Done()
-					if err := work(); err != nil {
-						i.resultOnce.Do(func() {
-							i.result = err
-						})
-					}
-				}()
+				i.exec(work)
```

Comment:
Can this just use the same logic, and pass in a context that never get cancelled?

Reply:
Done.